L1Trigger/TrackFindingTracklet/test/AnalyzerTracklet.cc

```diff
@@ -0,0 +1,247 @@
+    void beginJob() override {}
+    void beginRun(const Run& iEvent, const EventSetup& iSetup) override;
+    void analyze(const Event& iEvent, const EventSetup& iSetup) override;
+    void endRun(const Run& iEvent, const EventSetup& iSetup) override {}
+    void endJob() override;
+
+  private:
+    //
+    void associate(const vector<vector<TTStubRef>>& tracks, const StubAssociation* ass, set<TPPtr>& tps, int& sum) const;
+
+    // ED input token of tracks
+    EDGetTokenT<TTTracks> edGetToken_;
+    // ED input token of TTStubRef to selected TPPtr association
+    EDGetTokenT<StubAssociation> edGetTokenSelection_;
+    // ED input token of TTStubRef to recontructable TPPtr association
+    EDGetTokenT<StubAssociation> edGetTokenReconstructable_;
+    // Setup token
+    ESGetToken<Setup, SetupRcd> esGetTokenSetup_;
+    // DataFormats token
+    ESGetToken<DataFormats, DataFormatsRcd> esGetTokenDataFormats_;
+    // stores, calculates and provides run-time constants
+    const Setup* setup_;
+    // helper class to extract structured data from TTDTC::Frames
+    const DataFormats* dataFormats_;
+    // enables analyze of TPs
+    bool useMCTruth_;
+    //
+    int nEvents_;
+
+    // Histograms
+
+    TProfile* prof_;
+    TProfile* profChannel_;
+    TH1F* hisChannel_;
+    TH1F* hisEff_;
+    TH1F* hisEffTotal_;
+    TEfficiency* eff_;
+
+    // printout
+    stringstream log_;
+  };
+
+  AnalyzerTracklet::AnalyzerTracklet(const ParameterSet& iConfig) : useMCTruth_(iConfig.getParameter<bool>("UseMCTruth")), nEvents_(0) {
+    usesResource("TFileService");
+    // book in- and output ED products
+    const InputTag& inputTag = iConfig.getParameter<InputTag>("InputTag");
+    edGetToken_ = consumes<TTTracks>(inputTag);
+    if (useMCTruth_) {
+      const auto& inputTagSelecttion = iConfig.getParameter<InputTag>("InputTagSelection");
+      const auto& inputTagReconstructable = iConfig.getParameter<InputTag>("InputTagReconstructable");
+      edGetTokenSelection_ = consumes<StubAssociation>(inputTagSelecttion);
+      edGetTokenReconstructable_ = consumes<StubAssociation>(inputTagReconstructable);
+    }
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+    // log config
+    log_.setf(ios::fixed, ios::floatfield);
+    log_.precision(4);
+  }
+
+  void AnalyzerTracklet::beginRun(const Run& iEvent, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    // helper class to extract structured data from TTDTC::Frames
+    dataFormats_ = &iSetup.getData(esGetTokenDataFormats_);
+    // book histograms
+    Service<TFileService> fs;
+    TFileDirectory dir;
+    dir = fs->mkdir("Tracklet");
+    prof_ = dir.make<TProfile>("Counts", ";", 9, 0.5, 9.5);
```

Comment:
Either change histo title to CountsPerNonant, or add comment making clear that all your histograms are per nonant (if true).